### ISSUE TITLE
Newtonsoft.Json types replaced in Microsoft.AspNetCore.Authentication APIs

### ISSUE THREAD

**Issue by kinosang** [aspnet-contrib/AspNet.Security.OAuth.Providers#275]
aspnet/Announcements#341

> For derived OAuth implementations the most common change is to replace JObject.Parse with JsonDocument.Parse in the CreateTicketAsync override as shown [here](https://github.com/aspnet/AspNetCore/pull/7105/files?utf8=%E2%9C%93&diff=unified&w=1#diff-e1c9f9740a6fe8021020a6f249c589b0L40). Note JsonDocument is IDisposable.

**Comment by martincostello** [comment]
Updating to 3.0 preview 3, the auth providers now fail with an exception such as:

```
System.MissingMethodException: Method not found: 'Void Microsoft.AspNetCore.Authentication.OAuth.OAuthCreatingTicketContext..ctor(System.Security.Claims.ClaimsPrincipal, Microsoft.AspNetCore.Authentication.AuthenticationProperties, Microsoft.AspNetCore.Http.HttpContext, Microsoft.AspNetCore.Authentication.AuthenticationScheme, Microsoft.AspNetCore.Authentication.OAuth.OAuthOptions, System.Net.Http.HttpClient, Microsoft.AspNetCore.Authentication.OAuth.OAuthTokenResponse, Newtonsoft.Json.Linq.JObject)'.
   at AspNet.Security.OAuth.Amazon.AmazonAuthenticationHandler.CreateTicketAsync(ClaimsIdentity identity, AuthenticationProperties properties, OAuthTokenResponse tokens)
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start[TStateMachine](TStateMachine& stateMachine)
   at AspNet.Security.OAuth.Amazon.AmazonAuthenticationHandler.CreateTicketAsync(ClaimsIdentity identity, AuthenticationProperties properties, OAuthTokenResponse tokens)
   at Microsoft.AspNetCore.Authentication.OAuth.OAuthHandler`1.HandleRemoteAuthenticateAsync()
   at Microsoft.AspNetCore.Authentication.RemoteAuthenticationHandler`1.HandleRequestAsync()
```

**Comment by martincostello** [comment]
Have started a migration for some of the providers in #280.

**Comment by martincostello** [comment]
Will be resolved by #280.

**Comment by martincostello** [comment]
Closing in favour of #324.
